### PR TITLE
Add digitalocean cluster login

### PR DESCRIPTION
**Pull request by Typositoire**
- Add `doctl`
- Add source options for DO.

Available for testing:
```
- name: helm
    type: registry-image
    source:
      repository: typositoire/concourse-helm3-resource
      tag: 1.16.0-rc4
```

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -16,6 +16,11 @@ RUN for i in $(echo $HELM_PLUGINS | xargs -n1); do helm plugin install $i; done
 RUN curl -s "https://raw.githubusercontent.com/kubernetes-sigs/kustomize/master/hack/install_kustomize.sh"  | bash && \
   install kustomize /usr/local/bin/kustomize
 
+ARG DOCTL_VERSION=1.57.0
+RUN curl -sL -o /tmp/doctl.tar.gz https://github.com/digitalocean/doctl/releases/download/v${DOCTL_VERSION}/doctl-${DOCTL_VERSION}-linux-amd64.tar.gz && \
+  tar zxvf /tmp/doctl.tar.gz -C /usr/local/bin && \
+  chmod +x /usr/local/bin/doctl
```
Comment (avanier) [inline Dockerfile:22]: Is this necessary? The permissions should be kept within the tarball.

File: Dockerfile [hunk 2]
```diff
@@ -16,6 +16,11 @@ RUN for i in $(echo $HELM_PLUGINS | xargs -n1); do helm plugin install $i; done
 RUN curl -s "https://raw.githubusercontent.com/kubernetes-sigs/kustomize/master/hack/install_kustomize.sh"  | bash && \
   install kustomize /usr/local/bin/kustomize
 
+ARG DOCTL_VERSION=1.57.0
+RUN curl -sL -o /tmp/doctl.tar.gz https://github.com/digitalocean/doctl/releases/download/v${DOCTL_VERSION}/doctl-${DOCTL_VERSION}-linux-amd64.tar.gz && \
+  tar zxvf /tmp/doctl.tar.gz -C /usr/local/bin && \
```
Comment (avanier) [inline Dockerfile:21]: It's technically bad practice to mix dashed and non-dashed flags when invoking `tar`.

A safer invocation _(even if currently works as-in)_, would be :

```shell
tar -C /usr/local/bin -zxvf /tmp/doctl.tar.gz && \

**Review by avanier — APPROVED** [review]
LGTM, but I must say it wasn't the purpose of this resource to originally fetch `kubeconfig`s from external sources. This seems like a slippery slope. I don't see how it could be better done for now, but with the resourceV2 API, this hopefully should be better split in the future.

File: Dockerfile (hunk 1 above)
Comment (Typositoire) [reply Dockerfile:22]: The binary is not executable on untar xD

**Comment by Typositoire** [comment]
@avanier This is the ONLY way to authenticate to DigitalOcean, a bit like ECR vs DockerHub. You need to go through `doctl` because the certificate expires every 7 days which would make this a hell to manage without `doctl`.

**Comment by avanier** [comment]
> @avanier This is the ONLY way to authenticate to DigitalOcean, a bit like ECR vs DockerHub. You need to go through `doctl` because the certificate expires every 7 days which would make this a hell to manage without `doctl`.

Yeah, no I get that. My point is only that in the future with resourceV2 this should be revisited. It would make more sense to have a first thing get the kubecon... oh wait I'm writing it out and even that is icky. Ewwww... secrets at rest in a volume passed between resources. :vomiting_face: 

Yup, this is the least worse option. Move along citizen.

**Review by avanier — APPROVED** [review]

**Comment by Typositoire** [comment]
@avanier Oh I see what you mean... pass a `kubeconfig` path instead of fetching it with `doctl`. Yeah probably but like you see it's low effort to add xD and simplifies everything lol

**Pull request merged**